dot/rpc/subscription/listeners.go

```diff
@@ -16,20 +16,28 @@
 package subscription
 
 import (
-	"context"
+	"errors"
 	"fmt"
 	"reflect"
+	"time"
 
 	"github.com/ChainSafe/gossamer/dot/rpc/modules"
 	"github.com/ChainSafe/gossamer/dot/state"
 	"github.com/ChainSafe/gossamer/dot/types"
 	"github.com/ChainSafe/gossamer/lib/common"
 )
 
+var (
+	// ErrCannotCancel when is not possible to cancel a goroutine after `cancelTimeout` seconds
+	ErrCannotCancel = errors.New("cannot cancel listen goroutines")
+
+	cancelTimeout = time.Second * 10
```

Comment:
you can make this a default timeout, but actually include the `time.Duration` as an attribute in the struct.  In the constructor for the struct you can assign the default timeout.  That way you can override it easily per struct in the tests, so we don't need to wait 10s in the unit tests.

Reply:
Done